code/modules/chemistry/Reagents-Drugs.dm

```diff
@@ -492,6 +492,138 @@ datum
 				..()
 				return
 
+		drug/caffeine        //Unified chem for lots of caffeinated drinks, similar to how ethanol functions
+			name = "caffeine"
+			id = "caffeine"
+			description = "An addictive stimulant contained in coffee beans and many caffeinated beverages."
+			reagent_state = LIQUID
+			fluid_r = 230
+			fluid_g = 220
+			fluid_b = 230
+			addiction_prob = 1 //It lasts a while, it's not as low as it seems
+			addiction_min = 100
+			max_addiction_severity = "LOW"
+			depletion_rate = 0.1
+			taste = "bitter"
+			overdose = 60
+			threshold = THRESHOLD_INIT
+			var/heart_failure_counter = 0
+
+			proc/caffeine_message(var/mob/M, var/power) // condensing all the messages in the same place
+				if (ON_COOLDOWN(M, "Caffeine Message", 10 SECONDS)) //Reduce chat spam
+					return
+				switch(power)
+					if(1)
+						boutput(M, pick("<span class='notice'>You feel refreshingly energised.</span>",\
```

Comment:
the player probably only needs to know about the caffeine they've drank if it's killing them

Reply:
Makes sense, done